File: src/services/Catalogo/Catalogo.API/Data/Queries/CatalogoTodosQuery.cs
using System.ComponentModel.DataAnnotations;

namespace Catalogo.API.Data.Queries
{
  public record CatalogoTodosQuery
  {
    [Range(1, int.MaxValue, ErrorMessage = "page: mínimo {1}, máximo {2}.")]
    public int page { get; set; }

    [Range(5, 50, ErrorMessage = "limit: mínimo {1}, máximo {2}.")]
    public int limit { get; set; }

    [Range(0, 3, ErrorMessage = "order fora do range, 0 a 3.")]
    public ECatalogoTodosQueryOrder order { get; set; }

    public bool inStock { get; set; }

    public bool outOfStock { get; set; }
  }

  public enum ECatalogoTodosQueryOrder
  {
    NameAsc,
    NameDesc,
    PriceHighToLow,
    PriceLowToHigh
  }
}

File: src/mobile/lib/app/utils/queries/catalogo_todos_query.dart
class CatalogoTodosQuery {
  CatalogoTodosQuery({
    required this.page,
    required this.limit,
    required this.order,
    required this.inStock,
    required this.outOfStock,
    this.nome,
  });
  late final int page;
  late final int limit;
  late final int order;
  late final bool inStock;
  late final bool outOfStock;
  late final String? nome;

  Map<String, dynamic> toMap() {
    return {
      'page': page,
      'limit': limit,
      'order': order,
      'inStock': inStock,
      'outOfStock': outOfStock,
      'nome': nome,
    };
  }
}
